### PR TITLE
Adds optional post-share button to the standard Share Action.

### PR DESCRIPTION
**Pull request by DFurnes**
### What does this PR do?
This pull request adds the ability to have a post-share button on the standard "Share Action" block, just like we did with the SMS Share prototype. Unlike in the prototype though, we can use this on _any_ Share Action now (on mobile, desktop, you name it!)

screen shot 2018-03-05 at 4 41 12 pm

It also fixes what turns out was a long-standing issue where _all_ share affirmations would show the content from the first share on the given campaign, instead of showing the corresponding content for that particular share block.

### Any background context you want to provide?
I've kept these in `additionalContent` for this PR because I'd like some feedback on how best to model this kind of thing in Contentful. Two separate "Affirmation Button" and "Affirmation Link" fields?

Another idea I was toying with was a "Block" relationship that'd let us just display an optional block alongside this affirmation message... adds a bit more admin overhead, but could be handy!

### What are the relevant tickets/cards?
[#155551584](https://www.pivotaltracker.com/story/show/155551584)

### Checklist
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added appropriate feature/unit tests.
- [ ] Added screenshot to PR description of related front-end updates on **small** screens.
- [ ] Added screenshot to PR description of related front-end updates on **medium** screens.
- [ ] Added screenshot to PR description of related front-end updates on **large** screens.

### CODE REVIEW

File: resources/assets/components/ShareAction/ShareAction.js [hunk 1]
```diff
@@ -68,10 +69,17 @@ const ShareAction = (props) => {
...
+  confirmationAction: null,
+  confirmationActionLink: null,
   content: null,
 };
 
 ShareAction.propTypes = {
+  id: PropTypes.string.isRequired,
+  // affirmation: PropTypes.string,
+  // confirmationAction: PropTypes.string,
+  // confirmationActionLink: PropTypes.string,
```
Comment (mendelB) [inline resources/assets/components/ShareAction/ShareAction.js]: Are these pending us rendering an actual `<Modal>` post share?

Comment (DFurnes) [reply resources/assets/components/ShareAction/ShareAction.js]: Oof, yeah these were from experimenting with that but I should've removed 'em!

**Comment by weerd** [comment]
> Another idea I was toying with was a "Block" relationship that'd let us just display an optional block alongside this affirmation message... adds a bit more admin overhead, but could be handy!

Not sure if I'm understanding correctly...

So part of the issue is that there's an `affirmation` field in the `ShareAction`, and you're debating on names for housing the possibility of a `confirmationAction` and `confirmationActionLink` fields along with the message, correct?

And you mention the possibility of instead using a `blocks` reference field to display a block along with the message?

What if we got rid of the `affirmation` field in the `ShareAction`, added the `blocks` ref field instead, and then spruced up the `affirmation` content-type and what you reference is an `affirmation` block? This way they could be used in multiple spots on the same campaign, etc and ideally all the affirmation related info is contained in that instance of the `affirmation` content-type.

**Comment by DFurnes** [comment]
OH MY GOD, DUH! I forgot we have a _literal_ Affirmation Block! Of course!! 👍 👍 

Alright, I'm very in favor of using a block reference field here then. I'll see about hooking that up next, and we can write a ticket to migrate this text `affirmation` field into a real block in an upcoming sprint.

**Pull request merged**